### PR TITLE
Messages cleanup

### PR DESCRIPTION
**Pull request by nacho692**
While removing unused abstractions and methods an existing bug popped up.

https://github.com/rsksmart/rskj/blob/34b3fa789f8068628fdbccb9fa9d3719abe36bc6/rskj-core/src/main/java/org/ethereum/sync/SyncPool.java#L322-L331

Clearly SyncPool was not dropping peers. I did a couple of synchronization tests removing false and found out that at the default 30 seconds timeout peers were dropping and reconnecting regularly.

Increasing the time made it so that only peers that stopped sending messages get dropped.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/RskContext.java [hunk 1]
```diff
@@ -1461,14 +1462,16 @@ private StatusResolver getStatusResolver() {
 
     private RskWireProtocol.Factory getRskWireProtocolFactory() {
         if (rskWireProtocolFactory == null) {
-            rskWireProtocolFactory = () -> new RskWireProtocol(
+            rskWireProtocolFactory = (MessageQueue messageQueue, Channel channel) -> new RskWireProtocol(
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/RskContext.java]: I would leave the types out, they can be implicit

File: rskj-core/src/main/java/co/rsk/net/BlockSyncService.java [hunk 2]
```diff
@@ -214,7 +214,8 @@ private void requestMissingHash(MessageChannel sender, Keccak256 hash) {
 
         logger.trace("Missing block {}", hash.toString().substring(0, 10));
 
-        sender.sendMessage(new GetBlockMessage(hash.getBytes()));
+        final GetBlockMessage message = new GetBlockMessage(hash.getBytes());
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/net/BlockSyncService.java]: Please undo at commit 6a9f1b5e41774001845bf8c3e63c3ea14fce36a8

File: rskj-core/src/main/java/co/rsk/net/NodeBlockProcessor.java [hunk 3]
```diff
@@ -104,7 +104,8 @@ public void processNewBlockHashesMessage(@Nonnull final MessageChannel sender, @
                 .filter(b -> !hasBlock(b.getBytes()))
                 .forEach(
                         b -> {
-                            sender.sendMessage(new GetBlockMessage(b.getBytes()));
+                            final GetBlockMessage message1 = new GetBlockMessage(b.getBytes());
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/net/NodeBlockProcessor.java]: Please undo this and all below at commit 6a9f1b5e41774001845bf8c3e63c3ea14fce36a8

File: rskj-core/src/test/java/co/rsk/net/NodeBlockProcessorTest.java [hunk 4]
```diff
@@ -1072,7 +1073,7 @@ public void failIfProcessBlockHeadersRequestCountHigher()  {
         final Integer size = syncConfiguration.getChunkSize() + 1;
         processor.processBlockHeadersRequest(sender, 1, block.getHash().getBytes(), size);
 
-        verify(sender, never()).sendMessage(any());
+        verify(sender, never()).sendMessage(new RskMessage(any()));
```
Comment (colltoaction) [inline rskj-core/src/test/java/co/rsk/net/NodeBlockProcessorTest.java]: Undo at this commit

File: rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java [hunk 5]
```diff
@@ -50,64 +53,69 @@
...
      * also, is useful when returned BLOCK_BODIES msg doesn't cover all sent hashes
      * or in case when peer is disconnected
      */
     private final PeerScoringManager peerScoringManager;
-    protected final SyncStatistics syncStats = new SyncStatistics();
-    protected EthState ethState = EthState.INIT;
-    protected SyncState syncState = SyncState.IDLE;
-    protected boolean syncDone = false;
+    private final SyncStatistics syncStats = new SyncStatistics();
+    private Channel channel;
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java]: Make everything `final` if possible

File: rskj-core/src/main/java/org/ethereum/net/server/Channel.java [hunk 6]
```diff
@@ -238,10 +227,10 @@ public boolean equals(Object o) {
 
         Channel channel = (Channel) o;
 
-        if (inetSocketAddress != null ? !inetSocketAddress.equals(channel.inetSocketAddress) : channel.inetSocketAddress != null) {
+        if (!Objects.equals(inetSocketAddress, channel.inetSocketAddress)) {
```
Comment (colltoaction) [inline rskj-core/src/main/java/org/ethereum/net/server/Channel.java]: Good use of `Objects.equals`. Please join these into `return ... && ...;`

File: rskj-core/src/main/java/org/ethereum/listener/EthereumListener.java [hunk 7]
```diff
@@ -57,12 +57,6 @@
 
     void onTransactionPoolChanged(TransactionPool transactionPool);
 
-    /**
-     * @deprecated Check Rsk.hasBetterBlockToSync() and isPlayingBlocks()
```
Comment (colltoaction) [inline rskj-core/src/main/java/org/ethereum/listener/EthereumListener.java]: 😃

File: rskj-core/src/main/java/co/rsk/net/MessageChannel.java [hunk 8]
```diff
@@ -18,13 +18,12 @@
 
 package co.rsk.net;
 
-import co.rsk.net.eth.RskMessage;
 import co.rsk.net.messages.Message;
 
 import java.net.InetAddress;
 
 public interface MessageChannel {
-    void sendMessage(RskMessage rskMessage);
```
Comment (colltoaction) [inline rskj-core/src/main/java/co/rsk/net/MessageChannel.java]: You did this change and now you're undoing it. I suggest squashing this commit with 6a9f1b5e41774001845bf8c3e63c3ea14fce36a8

**Comment by nacho692** [comment]
closes #1004

**Pull request merged**